webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFact.scala

```diff
@@ -1042,6 +1099,54 @@ case class SelectFactConfig[A](
   def invertMode: SelectFactConfig[A] = if (this.mode == SelectMode.Ignore) toRetrieve else toIgnore
 
   override def toString: String = this.mode.toString
+
+  def merge(other: SelectFactConfig[A], fix: SelectMode): SelectFactConfig[A] = {
+    (this.mode, other.mode) match {
+      case (a, b) if (a == b) => this
+      case _                  => this.copy(mode = fix)
+    }
+  }
+
+  def max(other: SelectFactConfig[A]): SelectFactConfig[A] = merge(other, SelectMode.Retrieve)
```

Comment:
this is needed to accurratly compute the resulting `SelectFact` of the composed event of storing a node